components/brave_sync/brave_sync_service_impl.cc

```diff
@@ -489,6 +493,10 @@ void BraveSyncServiceImpl::OnResolvedPreferences(const RecordsList& records) {
     ResetSyncInternal();
   if (contains_only_one_device)
     OnResetSync();
+
+  if (had_less_than_two && have_two_or_more_after_merge) {
```

Comment:
per https://github.com/brave/brave-core/pull/1188#pullrequestreview-188147327
we should `StopLoop` first 
And should be sufficient checking one condition, see the comment above

Reply:
I think, we cannot skip send `StopLoop` because `base::RepeatingTimer` has a logic to set a new interval for existing timer:
https://cs.chromium.org/chromium/src/base/timer/timer.cc?type=cs&g=0&l=120
https://cs.chromium.org/chromium/src/base/timer/timer.cc?type=cs&g=0&l=137